src/lib.rs

```diff
@@ -234,20 +234,45 @@ pub fn clean(src: &str) -> String {
 /// [`id_prefix`]: #method.id_prefix
 /// [`tags`]: #method.tags
 /// [`clean_content_tags`]: #method.clean_content_tags
-#[derive(Debug)]
 pub struct Builder<'a> {
     tags: HashSet<&'a str>,
     clean_content_tags: HashSet<&'a str>,
     tag_attributes: HashMap<&'a str, HashSet<&'a str>>,
     generic_attributes: HashSet<&'a str>,
     url_schemes: HashSet<&'a str>,
     url_relative: UrlRelative,
+    url_filter_map_callback: Option<Box<UrlFilterCallback>>,
     link_rel: Option<&'a str>,
     allowed_classes: HashMap<&'a str, HashSet<&'a str>>,
     strip_comments: bool,
     id_prefix: Option<&'a str>,
 }
 
+type UrlFilterCallback = for<'a> Fn(&str, &str, &'a str) -> Option<Cow<'a, str>> + Send + Sync + 'static;
+
+impl<'a> fmt::Debug for Builder<'a> {
```

Comment:
Better idea: write a wrapper type around `Box<UrlFilterCallback>` that implements `Debug` itself, so that the main `Builder` class can stick to the derived implementation.

Reply:
Unfortunately that won't work due to orphan rules (it's an alias, not a real type).